RealmSwift/Tests/ObjectTests.swift

```diff
@@ -968,6 +1378,62 @@ class ObjectTests: TestCase {
         token.invalidate()
     }
 
+    func testOptionalPropertyKeyPathNotifications() {
+        let realm = try! Realm()
+        let object = SwiftOptionalDefaultValuesObject()
+        try! realm.write({
+            realm.add(object)
+        })
+
+        // Expect notification for change on observed path
+        var token = object.observe(keyPaths: ["optIntCol"], expectChange("optIntCol", 1, 2))
+        dispatchSyncNewThread {
+            let realm = try! Realm()
+            try! realm.write({
+                realm.objects(SwiftOptionalDefaultValuesObject.self).first!.optIntCol.value = 2
+            })
+        }
+        realm.refresh()
+        waitForExpectations(timeout: 0)
+        token.invalidate()
+
+        // Expect no notification for change outside of observed path
+        token = object.observe(keyPaths: ["optStringCol"], expectChange("optIntCol", 2, 3, true)) // Passing true inverts expectation
+        dispatchSyncNewThread {
+            let realm = try! Realm()
+            try! realm.write({
```

Comment:
In general, unless the compiler is getting confused (usually only happens in a `guard`), do not wrap your blocks with parentheses in swift.